src/software/backend/input/replay_logging/replay_reader.cpp

```diff
@@ -0,0 +1,139 @@
+#include "replay_reader.h"
+
+#include <google/protobuf/util/delimited_message_util.h>
+
+#include <fstream>
+
+// unfortunately protobuf forces me to write caveman code
+extern "C"
+{
+#include <fcntl.h>
+#include <unistd.h>
+}
+
+namespace fs = std::experimental::filesystem;
+
+namespace
+{
+    TbotsReplay readDelimitedReplayProtobufFile(const fs::path& file_path)
+    {
+        // imagine having to write caveman code to read protobuf msgs from files
```

Comment:
not a blocking comment, but why did we have to write "caveman code" for this instead of using `std::experimental::filesystem`?

Reply:
this is how I saw it in a guide online, seems like this is what most people are doing. however, this can also be implemented with a `IstreamInputStream` constructed from an `std::ifstream`.

Reply:
If it's not significantly more complex I think that'd be preferable. If it is more complex and so we elect to keep this as-is, a more descriptive comment about why we wrote it "caveman" style would be helpful :stuck_out_tongue:

Reply:
removed caveman code, using istream now